WordPressAuthenticator/Unified Auth/View Related/Sign up/SignupMagicLinkViewController.swift

```diff
@@ -0,0 +1,197 @@
+}
+
+
+// MARK: - Private Methods
+private extension SignupMagicLinkViewController {
+    /// Registers all of the available TableViewCells.
+    ///
+    func registerTableViewCells() {
+        let cells = [
+            GravatarEmailTableViewCell.reuseIdentifier: GravatarEmailTableViewCell.loadNib(),
+            TextLabelTableViewCell.reuseIdentifier: TextLabelTableViewCell.loadNib()
+        ]
+
+        for (reuseIdentifier, nib) in cells {
+            tableView.register(nib, forCellReuseIdentifier: reuseIdentifier)
+        }
+    }
+
+    /// Describes how the tableView rows should be rendered.
+    ///
+    func loadRows() {
+        rows = [.persona, .instructions, .checkSpam]
+
+        if let errorText = errorMessage, !errorText.isEmpty {
+            rows.append(.errorMessage)
+        }
+    }
+
+    /// Configure cells.
+    ///
+    func configure(_ cell: UITableViewCell, for row: Row, at indexPath: IndexPath) {
+        switch cell {
+        case let cell as GravatarEmailTableViewCell where row == .persona:
+            configureGravatarEmail(cell)
+        case let cell as TextLabelTableViewCell where row == .instructions:
+            configureInstructionLabel(cell)
+        case let cell as TextLabelTableViewCell where row == .checkSpam:
+            configureCheckSpamLabel(cell)
+        case let cell as TextLabelTableViewCell where row == .errorMessage:
+            configureErrorLabel(cell)
+        default:
+            DDLogError("Error: Unidentified tableViewCell type found.")
+        }
+    }
+
+    /// Configure the gravatar + email cell.
+    ///
+    func configureGravatarEmail(_ cell: GravatarEmailTableViewCell) {
+        cell.configure(withEmail: loginFields.username)
+    }
+
+    /// Configure the instruction cell.
+    ///
+    func configureInstructionLabel(_ cell: TextLabelTableViewCell) {
+        cell.configureLabel(text: WordPressAuthenticator.shared.displayStrings.openMailSignupInstructions, style: .body)
+    }
+
+    /// Configure the "Check spam" cell.
+    ///
+    func configureCheckSpamLabel(_ cell: TextLabelTableViewCell) {
+        cell.configureLabel(text: WordPressAuthenticator.shared.displayStrings.checkSpamInstructions, style: .body)
+    }
+
+    /// Configure the error message cell.
+    ///
+    func configureErrorLabel(_ cell: TextLabelTableViewCell) {
+        cell.configureLabel(text: errorMessage, style: .error)
+        if shouldChangeVoiceOverFocus {
+            UIAccessibility.post(notification: .layoutChanged, argument: cell)
+        }
+    }
+
+    // MARK: - Private Constants
+
+    /// Rows listed in the order they were created.
+    ///
+    enum Row {
+        case persona
+        case instructions
+        case checkSpam
+        case errorMessage
+
+        var reuseIdentifier: String {
+            switch self {
+            case .persona:
+                return GravatarEmailTableViewCell.reuseIdentifier
+            case .instructions, .checkSpam, .errorMessage:
```

Comment:
Is this formatting used in WPiOS? If not I can change it back to individual cases:
```
case .instructions:
case .checkSpam:
case .errorMessage:
```

Reply:
I think one `case` is fine. And probably preferred (at least, by me 😄 ).